components/org.wso2.carbon.stream.processor.core/src/main/java/org/wso2/carbon/stream/processor/core/internal/StreamProcessorService.java

```diff
@@ -115,19 +116,34 @@ public void deploySiddhiApp(String siddhiAppContent, String siddhiAppName) throw
                 if (haManager.isActiveNode()) {
                     //Active Node
                     if (StreamProcessorDataHolder.isPersistenceEnabled()) {
-                        log.info(
-                                "Periodic Persistence of Active Node Enabled. Restoring From Last Saved Snapshot " +
-                                        "for " + siddhiAppName);
-                        String revision = null;
-                        try {
-                            revision = siddhiAppRuntime.restoreLastRevision();
-                        } catch (CannotRestoreSiddhiAppStateException e) {
-                            log.error("Error in restoring Siddhi app " + siddhiAppRuntime.getName(), e);
-                        }
-                        if (revision != null) {
-                            log.info("Siddhi App " + siddhiAppName + " restored to revision " + revision);
-                        }
 
+                        if (persistenceStoreClearEnabled) {
+
+                            if (siddhiApp != null && siddhiApp.equals(siddhiAppName)) {
+                                siddhiAppRuntime.clearAllRevisionsOfSiddhiAppInPersistenceStore();
+                                isSingleAppPersistenceClear = true;
+                                log.info("Deleting all the revisions of the Periodic Persistence of " +
+                                        "Active Node for " + siddhiAppName);
+                            } else if (!isSingleAppPersistenceClear) {
```

Comment:
through worker.sh, we can passed one siddhi app to clear its persistence store. So if we pass as one application, 'isSingleAppPersistenceClear' would be true

Reply:
But you are setting it inside the above if. So what is the need of having it here. You can just say else

Reply:
when it is in else part, see the below example;

let's say we have 3 siddhi apps,
 1. testSiddhiApp1
 2. testSiddhiApp2
 3. testSiddhiApp3

and we invoke the worker.sh as,

./worker.sh -DpersistenceStoreClearEnabled=true  -DsiddhiApp=testSiddhiApp1

when it goes through StreamProcessorService.deploySiddhiApp() method, according to your scenario;

`
                            if (siddhiApp != null && siddhiApp.equals(siddhiAppName)) {
                                siddhiAppRuntime.clearAllRevisions();
                                log.info("Deleting all the revisions of the Periodic Persistence of " +
                                        "Active Node for " + siddhiAppName);

                            } else {
                                log.info("Deleting all the revisions of the Periodic Persistence of " +
                                        "Active Node for " + siddhiAppName);
                                siddhiAppRuntime.clearAllRevisions();
                            }
`

it will execute the if block, but when it comes 2nd siddhi app to deploy, it will go and execute else block. but we just need to clear revisions of testSiddhiApp1 only

Reply:
No. Then your code should look like below. Current code has a bug where testSiddhiApp3 get deployed before testSiddhiApp1 it's state will be deleted.
```
if (persistenceStoreClearEnabled) {
                            if (siddhiApp != null ) {
                            	if (siddhiApp.equals(siddhiAppName)) {
                            		siddhiAppRuntime.clearAllRevisionsOfSiddhiAppInPersistenceStore();
                                log.info("Deleting all the revisions of the Periodic Persistence of " +
                                        "Active Node for " + siddhiAppName);
                            	}
                            } else {
                            	siddhiAppRuntime.clearAllRevisionsOfSiddhiAppInPersistenceStore();
                                log.info("Deleting all the revisions of the Periodic Persistence of " +
                                        "Active Node for " + siddhiAppName);
                            }
                        } else {
                            ...
                        }
```

Reply:
done accordingly